analyzers/tests/SonarAnalyzer.Test/Helpers/MessageTemplatesTest.cs

```diff
@@ -0,0 +1,150 @@
+    public void Parse_Placeholder(string template, string placeholder, int start, int end)
+    {
+        var result = MessageTemplates.Parse(template);
+        ShouldBeSuccess(result, 1);
+        ShouldBe(result.Placeholders[0], placeholder, start, end);
+    }
+
+    [DataTestMethod]
+    // alignment
+    [DataRow("hello {world,1}")]
+    [DataRow("hello {world,42}")]
+    [DataRow("hello {world,-1}")]
+    [DataRow("hello {world,-199}")]
+    // format
+    [DataRow("hello {world:format}")]
+    [DataRow("hello {world:42}")]
+    [DataRow("hello {world:{}")]
+    [DataRow("hello {world:!@#$%^&*()_}")]
+    [DataRow("hello {world:42}")]
+    [DataRow("hello {world:#for_mat42}")]
+    // mixed
+    [DataRow("hello {world,1:format}")]
+    [DataRow("hello {world,42:!@#$%}")]
+    [DataRow("hello {world,-1:dd-MM-yyy}")]
+    [DataRow("hello {world,-42:3,14159}")]
+    public void Parse_Placeholder_Named_Alignment_Format(string template)
+    {
+        var result = MessageTemplates.Parse(template);
+        ShouldBeSuccess(result, 1);
+        ShouldBe(result.Placeholders[0], "world", 7, 5);
+    }
+
+    [TestMethod]
+    public void Parse_Placeholder_Multiple()
+    {
+        var template = """
+            In code's {$silent} realm,
+            Logic weaves through lines {@0}f text,
+            Errors {@teach,42} us well.
+
+            Syntax {sym_phony,42:dd-MM},
+            Logic {@_orchestrates42} the mind,
+            Programmer's {ballet:_}.
+            """;
+        var result = MessageTemplates.Parse(template);
+        ShouldBeSuccess(result, 6);
+        ShouldBe(result.Placeholders[0], "silent", 12, 6);
+        ShouldBe(result.Placeholders[1], "0", 56, 1);
+        ShouldBe(result.Placeholders[2], "teach", 75, 5);
+
+        ShouldBe(result.Placeholders[3], "sym_phony", 103, 9);
+        ShouldBe(result.Placeholders[4], "_orchestrates42", 132, 15);
+        ShouldBe(result.Placeholders[5], "ballet", 173, 6);
+    }
+
+    [DataTestMethod]
+
+    [DataRow("{")]                              // Left bracket is not allowed
+    [DataRow("{{{")]                            // Third left bracket is not allowed (first two are valid)
+    [DataRow("{}")]                             // Empty placeholder is not allowed
+    [DataRow("{{{}}}")]                         // Empty placeholder is not allowed
+    [DataRow("Login failed for {User")]         // Missing closing bracket
+    [DataRow("Login failed for {&User}")]       // Only '@' and '$' are allowed as prefix
+    [DataRow("Login failed for {User_%Name}")]  // Only alphanumerics and '_' are allowed for placeholders
+    [DataRow("Retry attempt {Cnt,r}")]          // The alignment specifier must be numeric
+    [DataRow("Retry attempt {Cnt,}")]           // Empty alignment specifier is not allowed
+    [DataRow("Retry attempt {Cnt:}")]           // Empty format specifier is not allowed
+    public void Parse_Placeholder_Failure(string template)
```

Comment:
...actually, it works 😩 
Because `format` matches everything except `}`, this is valid:

```csharp
"Retry attempt {Cnt:format,42}"
//                  ^^^^^^^^^ all of this is a format
```

Reply:
Well, it documents a supported scenario (by us; I don't know about the logging frameworks) that is not intuitive and it's not documented in our tests. Why not add it?

Reply:
Added it!